tests/Tests/UnitTests/EmbeddedResourceDataAttribute.cs

```diff
@@ -0,0 +1,45 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Reflection;
+using Xunit.Sdk;
+
+namespace Concordium.Sdk.Tests.UnitTests;
+
+/// <summary>
+/// <see cref="DataAttribute"/> used to specify data captured from resources embedded
+/// in the manifest. This is used for the theory in <see cref="Xunit"/> tests.
+/// </summary>
+public class EmbeddedResourceDataAttribute : DataAttribute
```

Comment:
IMHO this class is adding unnecessary complexity. When this attribute is used is only reads one file either way.

Just use `Theory` and then ready the file with `File.ReadXXX` in the test.

Reply:
Are saying it is possible to read a file embedded in the manifest from a path using `File.Read`? I can only find references that prescribe [Assembly.GetManifestResourceStream Method](https://learn.microsoft.com/en-us/dotnet/api/system.reflection.assembly.getmanifestresourcestream#System_Reflection_Assembly_GetManifestResourceStream_System_String_). It's quite nifty to have a way to do this, and it does not preclude future maintainers from doing what you prescribe - if that is possible.

I think it's probably better if you give the SDK an overhaul after I hand it over since a lot of the changes prescribed are quite throughout in nature, and some may be a matter of opinion.

Reply:
This works

```
    [Theory]
    [InlineData("./Wallets/Data/GenesisWalletKeyExportFormatMissingField.json")]
    [InlineData("./Wallets/Data/BrowserWalletKeyExportFormatMissingField.json")]
    public void FromWalletKeyExportFormat_OnInputWithMissingField_ThrowsException(string path)
    {
        var json = File.ReadAllText(path);
        Action result = () => WalletAccount.FromWalletKeyExportFormat(json);
        result.Should().Throw<WalletDataSourceException>();
    }
```